File: src/ScavengerHuntEditor-Backend/ScavengerHunt.Domain/Entities/Assignment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScavengerHunt.Domain.Entities
{
    public sealed class Assignment
    {
        public int Id { get; set; }

        public int HuntId { get; set; }
        public required Hunt Hunt { get; set; }

        public int HintId { get; set; }
        public required Hint Hint { get; set; }

        public int SolutionId { get; set; }
        public required Solution Solution { get; set; }

    }
}

File: src/fe-hunt-editor/src/lib/models/Assignment.ts
import type { Hint } from './Hint';
import type { Solution } from './Solution';

/**
 * Represents an assignment in a hunt.
 */
export interface Assignment {
	/** The unique identifier of the assignment. */
	id: number;
	/** The hint associated with the assignment. */
	hint: Hint;
	/** The solution associated with the assignment. */
	solution: Solution;
}
